src/app.ts

```diff
@@ -225,15 +225,19 @@ async function app() {
 
   logger.info(`Exported ${count} FHIR resources as JSON.`);
 
-  // If ig-data exists, generate an IG, otherwise, generate resources only
+  // If template is given in the config, generate an IG, otherwise, generate resources only
   let isIG = false;
   const igDataPath = path.resolve(input, 'ig-data');
-  if (fs.existsSync(igDataPath)) {
+  if (config.template) {
     isIG = true;
     logger.info('Assembling Implementation Guide sources...');
     const igExporter = new IGExporter(outPackage, defs, igDataPath, isIgPubContext);
     igExporter.export(outDir);
     logger.info('Assembled Implementation Guide sources; ready for IG Publisher.');
+  } else if (fs.existsSync(igDataPath)) {
+    logger.warn(
+      'No template property is specified in configuration, so no output specific to IG creation will be generated. Therefore the ig-data folder will be unused.'
+    );
```

Comment:
Interesting point. So there is a totally valid use case to have to `template` but to have an ig-data folder. Which I feel like leaves us with a few options:
* Get rid of this warning altogether
* Only throw this warning if there are files in ig-data that clearly are only relevant to building an ig, such as xml or markdown content
* Make this an info that says something like "No IG will be generated"

I think I kinda like option 3. Option 1 seems not informational enough, and option 2 seems like overkill for a warning that isn't particularly important. I think option 3 is good to have so that people know they are not getting IGExporter functionality. What do you think?

Reply:
I agree with your assessment of 1 and 2. I think I like 3! I was trying to think of how to word the warning without getting overly confusing and wordy, but 3 is nice and simple and informative.

Reply:
Well but also now that I think about this more deeply. I think I just need to change this PR now. We know that we are going to want the export of `ig.ini` to work such that if no `template` is specified, this indicates that we shouldn't generate an `ig.ini`, but this does not necessarily mean that we are not supposed to generate an ig.json, or a menu.xml, or so on and so forth. Maybe the person just doesn't want to make an `ig.ini`.

Reply:
Yeah, we can't / shouldn't use template as a flag for whole IG generation anymore.  I think that, for now, we go back to whether or not `ig-data` is present, or we introduce a new configuration property to control it.  Given the temporary nature of this all, and that relying on presence or absence of `ig-data` is already established in current versions of SUSHI, maybe we just keep it that way?  And if we find clearly IG-related fields (like `template`) but no `ig-data` folder, we log a warning?  Maybe?

Reply:
I agree, let's go back to `ig-data`. I think it is possible that someone could not have an `ig-data` and still technically intend to generate an IG, but in that case that IG would actually fail I think, because the person would not have an `index.md`.

Reply:
Or actually one potential alternative would be to introduce two new flags to the config. Something like `generateIndex` and `generateIG`. The `generateIndex` flag if set to true would indicate that the user wants us to generated `index.md` for them. If set to false or unset, it would indicate that they either have an `index.md` in their `ig-data` which we will be copying over, or they have one in their output which we should not overwrite. The `generateIG` flag would work similarly to how the `config.template` is working in this PR. This way @cmoesel in your task for converting existing configs, you could set both these flags to true, and existing IGs would still work, while we still give people the freedom to turn off automatic overwriting of `index.md` if they so choose. This does add some complexity to the config.yaml though.

Reply:
But the thing I like about adding those flags is then for (I think) every artifact we generate, there is a pattern that if something is set in the config.yaml, or if the thing is in `ig-data`, we put the thing in the output. If the relevant parameter is not in the config, then we do not generate the artifact and risk overwriting anything.

Reply:
The thing is, the generated index.md is pretty much worthless.  It's like one sentence of text.  Nobody would ever ever want to keep our generated index.md.  It was basically just there so that IG Pub would succeed.  So it seems unfortunate to have a flag for something that no one would ever want as a long-term (or even medium-term) solution...

Reply:
It's unfortunate, but it would retain the ability for us to give people one single file, tell them to define their FSH and away they go in terms of being able to run SUSHI and then the IG Pub using only that one file.

Reply:
It would align a little better w/ the others if it weren't a flag, but rather a data parameter.  So... what do you think of `indexPageContent`?  If it's empty, we do nothing and/or copy over the existing index.md/index.xml.  Otherwise (if it exists), we generate an index.md with the contents.  I still don't love it, but it seems to address the concerns for now.

Reply:
I think that's the best solution given our current constraints. Just a simple string I am assuming right?

This would also lead me to want to make `generateIG` an explicit parameter, not something that is implicit based off of `ig-data`.

Reply:
Would `generateIG` be a required parameter?  If not, is the default `true`?  If it is optional, would it be better for the parameter to be the negative use case (e.g., `fshOnly`)?

Reply:
My thinking would be not required, so I think you're right it is better as a negative use case.